community-tutorials/mail-on-ssh-login/01-en.md

```diff
@@ -0,0 +1,165 @@
+Existing customers can add the product easily and quickly.
+
+# Step 2 - Basic configuration of the server
+Ensure that the server is up to date and upgrade the system:
+```
+sudo apt update && sudo apt dist-upgrade -y
+```
+Install the required package ssmtp:
+```
+sudo apt install ssmtp -y
+```
+# Step 3 - Configuration of SSMTP daemon
+In this steep we will configure SSMTP to use a mail account to send mails from. Please create a new email address for this purpose. You will need the following data:
+
+- Username for the new created email address
+- Password for the new created email address
+- SMTP server for the new created email address
+- SMTP Port, should be 25.
+
+Create configuration file for SSMTP
+
+```
+sudo nano  /etc/ssmtp/ssmtp.conf
+```
+
+add the following lines but update the login information to work with your newly created mail account.
+
+```
+#/etc/ssmtp/ssmtp.conf
+root=server@YourDomainFromNetcup.com
+mailhub= mx2f11.netcup.net:25
+rewriteDomain=YourDomainFromNetcup.com
+AuthUser=YourDomainFromNetcup.com 
+AuthPass=YourPasswordForTheMailAccount
+FromLineOverride=YES
+```
+
+This will enbale SSMTP to send mails from your email account. It is recommended to create a mail account for this purpose only! Don’t use your daily account!
+
+# Step 4 – create a mail template
+The server is now able to write mails. To do that on a login, we must create a mail template.
+We could just send a simple alert, but it is nice to get more information to identify a possible thread.
+Therefore we will send a mail that contains the username of the user, the IP, date and time of the login. 
+
+Create a file for that purpose.
+`sudo nano /srv/mailatlogin.sh `
+
+Copy and paste this content to your file
+```
+#!/bin/sh
+date=$(date '+%d.%m.%Y')
+host=$(hostname)
+ip=$(hostname -I)
+{
+echo To: Reciever@yourDomain.com
+echo From: SERVER YourDomainFromNetcup.com
+echo Subject: "SERVER LOGIN! from $PAM_USER on $host IP: $PAM_RHOST $date"
+echo
+echo "Login detected on the server $HOSTNAME from 
+IP: $ip 
+User: $PAM_USER
+If this is not you, act immediately!"
+} | ssmtp Reciever@yourDomain.com
+```
+
+Save and exit the editor.
+The first lines reads the information from your system and stores them into variables for later use. We get the date, hostname and ip address. Then we will add this information to your mail.  
+All content between the curly braces is the mail.  
+Echo To  and Echo From must be supplemented with your data  
+Echo Subject: “ is the Subject of your mail. It can be edited if needed. All values starting with $ are variables. There will automatically be replaced with the corresponding value.  
+The next echo is the content of the mail and ca be edited or replaced.   
+Please be aware of the last line! Here you need to replace Reciever@yourDomain.com with your mailadress!
+# Step 5. Testing
+Test the script with the command
+```
+sudo /srv/mailatlogin.sh
```

Comment:
In order to execute this command I had to it with bash or shell.
```
sudo bash /srv/mailatlogin.sh
sudo sh /srv/mailatlogin.sh
```

Reply:
you are right, i forgot so make the script executable
chmod +x /srv/mailatlogin.sh
I'll add this to the tutorial